lib/PrintAsClang/DeclAndTypePrinter.cpp

```diff
@@ -436,6 +443,54 @@ class DeclAndTypePrinter::Implementation
         os << "    return *this == cases::";
         syntaxPrinter.printIdentifier(pair.first->getNameStr());
         os << ";\n  }\n";
+
+        if (!pair.first->hasAssociatedValues()) {
+          continue;
+        }
+
+        auto associatedValueList = pair.first->getParameterList();
+        // TODO: add tuple type support
+        if (associatedValueList->size() > 1) {
+          continue;
+        }
+        auto firstType = associatedValueList->front()->getType();
+        auto firstTypeDecl = firstType->getNominalOrBoundGenericNominal();
+        OptionalTypeKind optKind;
+        std::tie(firstType, optKind) =
+            getObjectTypeAndOptionality(firstTypeDecl, firstType);
+
+        // FIXME: (tongjie) may have to forward declare return type
+        os << "  inline ";
+        clangFuncPrinter.printClangFunctionReturnType(
+            firstType, optKind, firstTypeDecl->getModuleContext(),
+            owningPrinter.outputLang);
+        os << " get" << name << "() const {\n";
+        os << "    if (!is" << name << "()) abort();\n";
+        os << "    auto thisCopy = *this;\n";
```

Comment:
While this copy is correct, it incurs an extra destroy operation on the value, as you're using a C++ class value on stack, whose destructor gets called and incorrectly tries to destroy the enum after `_destructiveProjectEnumData`. We should avoid that. One idea is to use placement `new` on a buffer you allocate on stack that can store the copy C++ type. This way we can avoid calling the destructor there.